### DISCUSSION TITLE
How can i renderHTML a customNode with a async function

### DISCUSSION THREAD

**Discussion by adou2236** [ueberdosis/tiptap#1926] [General Questions]

**Comment by philippkuehn** [comment]
please add more context.

**Reply by adou2236** [reply]
my Node has some variable quantity ，i want translate them when i use editor.getHtml()

**Comment by philippkuehn** [comment]
renderHTML must be synchronous. you may want to add a [node view](https://tiptap.dev/guide/node-views) to your node.

**Reply by adou2236** [reply]
node view can't change the content if i want genetateHtml
